src/test/lib/concurrency/stress_test.cpp

```diff
@@ -587,104 +587,111 @@ TEST_F(StressTest, OperatorRegistration) {
   }
 }
 
-// For cases where issues usually arise early, we run the test multiple times.
-class StressTestMultipleRuns : public StressTest, public ::testing::WithParamInterface<int> {};
-
-INSTANTIATE_TEST_SUITE_P(StressTest, StressTestMultipleRuns, testing::Range(0, 10),
-                         [](const testing::TestParamInfo<StressTestMultipleRuns::ParamType>& info) {
-                           return "loop" + std::to_string(info.param);
-                         });
-
 /**
- * Test to verify that rollbacking Insert operations does not lead to multiple (outdated) rows being visible. This issue
- * has occurred when using link-time optimization or when inlining MVCC functions (see #2649).
- * We execute and immediately rollback insert operations in multiple threads. In parallel, threads are checking that no
+ * Test to verify that rolling back Insert operations does not lead to multiple (outdated) rows being visible. This
+ * issue has occurred when using link-time optimization or when inlining MVCC functions (see #2649).
+ * We execute and immediately roll back insert operations in multiple threads. In parallel, threads are checking that no
  * new rows are visible.
  */
-TEST_P(StressTestMultipleRuns, VisibilityOfRollbackedInserts) {
+TEST_F(StressTest, VisibilityOfInsertsBeingRolledBack) {
   // StressTestMultipleRuns runs 10x. Limit max runtime for *SAN builds.
   constexpr auto RUNTIME = std::chrono::seconds(5);
   constexpr auto MAX_VALUE_AND_ROW_COUNT = uint32_t{17};
   constexpr auto MAX_LOOP_COUNT = uint32_t{10'000};  // Experimentally determined, see #2651.
```

Comment:
Just to note that down: This takes around a minute in the sanitizer builds. Not necessarily a call for action, but to consider. Not sure how well we can trigger this in the slower sanitizer builds anyway.